### PR TITLE
Add length constraints for Person providers

### PR DESCRIPTION
**Pull request by MetcalfeTom**
### What does this changes

This PR works on https://github.com/joke2k/faker/issues/1506 by adding maximal and minimal length for person providers

Continuation of https://github.com/joke2k/faker/pull/1615

### CODE REVIEW

File: faker/providers/person/__init__.py [hunk 1]
```diff
@@ -205,11 +206,17 @@ def name(self) -> str:
...
         return self.generator.parse(pattern)
 
-    def first_name(self) -> str:
-        return self.random_element(self.first_names)
+    def first_name(self, min_length: Optional[int] = None, max_length: Optional[int] = None) -> str:
+        return self.random_element(self.first_names, min_length, max_length)
 
-    def last_name(self) -> str:
-        return self.random_element(self.last_names)
+    def first_name_as_list(self) -> Union[tuple, OrderedDict, dict]:
```
Comment (fcurella) [inline faker/providers/person/__init__.py:212]: I think we can get a little bit more specific withthe type hint here:
```suggestion
    def first_name_as_list(self) -> Union[Tuple[str, ...], OrderedDict[str, float], Dict[str, float]]:
```

Comment (MetcalfeTom) [reply faker/providers/person/__init__.py:212]: For all similar methods correct?  (e.g. `suffix_female_as_list()`)

Comment (fcurella) [reply faker/providers/person/__init__.py:212]: correct

Comment (MetcalfeTom) [reply faker/providers/person/__init__.py]: Done

**Comment by MetcalfeTom** [comment]
@fcurella can you advise me on the appveyor build error?
```
The build phase is set to "MSBuild" mode (default), but no Visual Studio project or solution files were found in the root directory. If you are not building Visual Studio project switch build mode to "Script" and provide your custom build command.
```

Also how do you want to proceed with the other providers?  Do they all need updating?  In which case, do you want me to cover them all in this PR or add them via separate PRs?

**Comment by fcurella** [comment]
@MetcalfeTom you can just ignore the Appveyor build.

> how do you want to proceed with the other providers? Do they all need updating?
Yes, they will all need to be updated.

> do you want me to cover them all in this PR or add them via separate PRs?

I'd rather have separate, smaller PRs.

**Comment by MetcalfeTom** [comment]
> I'd rather have separate, smaller PRs.

Can do.  How do you want to proceed?

You could open up a new branch based on `master`.  I can change the base of this PR to that and begin opening up others for the providers.  After those are complete we can merge into `master`

**Comment by fcurella** [comment]
> I can change the base of this PR to that and begin opening up others for the providers. After those are complete we can merge into master

I'd rather re-fork off of `master` each time for each PR. I think this way we could reduce merge conflicts along the way.

**Comment by MetcalfeTom** [comment]
@fcurella as directed, I opened up PRs based on all applicable provider types.  Please review the changes here when you have time, as it will be the first to merge

**Comment by MetcalfeTom** [comment]
@fcurella none of the providers in the library satisfy only this clause:

```python
if hasattr(self, "suffixes_male") and hasattr(self, "suffixes_female"):
    suffixes = tuple(self.suffixes_male) + tuple(self.suffixes_female)  # type: ignore[attr-defined]
    return suffixes
```

therefore I can't increase the test coverage of `faker/providers/person/__init__.py`

**Review by fcurella — APPROVED** [review]
Thank you!

**Comment by fcurella** [comment]
Sorry @MetcalfeTom , I had to revert all the length-check changes because the test suite started failing.

**Comment by MetcalfeTom** [comment]
@fcurella is there somewhere I can read more about the failing tests?  Trying to understand if I can help get this code back in

**Comment by fcurella** [comment]
you can look at [this job](https://github.com/joke2k/faker/actions/runs/3244082077/jobs/5319758523).

**Comment by MetcalfeTom** [comment]
@fcurella I can fix this by:

1. Adding additional elements to fit the length constraints
2. Changing the test criteria (seems arbitrary)

Probably option 1 is better.  Should I open a PR?

**Comment by fcurella** [comment]
Let's go with option #1

**Pull request merged**